Client/multiplayer_sa/CMultiplayerSA_DeviceSelection.cpp

```diff
@@ -0,0 +1,157 @@
+                result = queryDisplayConfig(QDC_ONLY_ACTIVE_PATHS, &pathCount, paths.get(), &modeCount, modes.get(), nullptr);
+            } while (result == ERROR_INSUFFICIENT_BUFFER);
+
+            if (result == ERROR_SUCCESS)
+            {
+                for (size_t i = 0; i < pathCount; i++)
+                {
+                    DISPLAYCONFIG_TARGET_DEVICE_NAME targetName = {};
+                    targetName.header.adapterId = paths[i].targetInfo.adapterId;
+                    targetName.header.id = paths[i].targetInfo.id;
+                    targetName.header.type = DISPLAYCONFIG_DEVICE_INFO_GET_TARGET_NAME;
+                    targetName.header.size = sizeof(targetName);
+                    const LONG targetNameResult = DisplayConfigGetDeviceInfo(&targetName.header);
+
+                    DISPLAYCONFIG_SOURCE_DEVICE_NAME sourceName = {};
+                    sourceName.header.adapterId = paths[i].sourceInfo.adapterId;
+                    sourceName.header.id = paths[i].sourceInfo.id;
+                    sourceName.header.type = DISPLAYCONFIG_DEVICE_INFO_GET_SOURCE_NAME;
+                    sourceName.header.size = sizeof(sourceName);
+                    const LONG sourceNameResult = DisplayConfigGetDeviceInfo(&sourceName.header);
+                    if (targetNameResult == ERROR_SUCCESS && sourceNameResult == ERROR_SUCCESS && targetName.monitorFriendlyDeviceName[0] != '\0')
+                    {
+                        char gdiDeviceName[std::size(sourceName.viewGdiDeviceName)];
+                        char monitorFriendlyDeviceName[std::size(targetName.monitorFriendlyDeviceName)];
+                        WideCharToMultiByte(CP_ACP, 0, sourceName.viewGdiDeviceName, -1, gdiDeviceName, static_cast<int>(std::size(gdiDeviceName)), nullptr,
+                                            nullptr);
+                        WideCharToMultiByte(CP_ACP, 0, targetName.monitorFriendlyDeviceName, -1, monitorFriendlyDeviceName,
+                                            static_cast<int>(std::size(monitorFriendlyDeviceName)), nullptr, nullptr);
+
+                        monitorNames.try_emplace(gdiDeviceName, monitorFriendlyDeviceName);
+                    }
+                }
+            }
+        }
+        FreeLibrary(user32Lib);
+    }
+
+    return monitorNames;
+#endif
+}
+
+struct RwSubSystemInfo
+{
+    char name[80];
+};
+
+#define FUNC_rwDeviceSystemRequest 0x7F2AB0
+using rwDeviceSystemRequest = RwSubSystemInfo*(__cdecl*)(void* device, uint32_t requestId, RwSubSystemInfo* pOut, void* pInOut, uint32_t numIn);
+static RwSubSystemInfo* RwEngineGetSubSystemInfo_Hooked(RwSubSystemInfo* subSystemInfo, int32_t subSystemIndex)
+{
+    static auto rwDeviceSystemRequestFunc = (rwDeviceSystemRequest)(FUNC_rwDeviceSystemRequest);
+    auto        devicePointer = *(uint32_t*)(0xC97B24);
+    auto        result = rwDeviceSystemRequestFunc((void*)(devicePointer + 0x10), 14, subSystemInfo, nullptr, subSystemIndex);
```

Comment:
whats that mem address (`devicePointer + 0x10`)?

Reply:
![obraz](https://github.com/user-attachments/assets/8cfbfbd3-491d-4db1-bb84-5ceb7ffce353)
Neither of those is declared in MTA, and only other places that do use it are just plain ASM.
Added the incomplete stubs for them in Renderware.h now, so it's a bit clearer